### PR TITLE
Use GITHUB_ACTION_PATH instead of github.action_path

### PR DESCRIPTION
**Pull request by flynneva**
This is required in order to use this action within a docker container.

Relates to #21

### CODE REVIEW

**Comment by flynneva** [comment]
Successful [example here](https://github.com/ros-drivers/usb_cam/actions/runs/4793706968/jobs/8526398937) using the action within a github action docker container strategy.  Script doesnt work yet for ubuntu 20.04, but does work for 22.04 👍🏼

File: action.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ runs:
 
     - id: copy-user-path
       run: |
-        sudo mkdir -p ${{github.action_path}} &&           \
+        sudo mkdir -p ${{ github.action_path }} &&           \
```
Comment (kbieganski) [inline action.yml:31]:
```suggestion
        sudo mkdir -p ${{ github.action_path }} &&         \
```

File: action.yml [hunk 2]
```diff
@@ -68,12 +68,12 @@ runs:
       shell: bash
 
     - id: create-pip-virtual-environment
-      run: cd ${{github.action_path}} && mkdir -p venv-dir && python3 -m venv venv-dir
+      run: cd ${GITHUB_ACTION_PATH} && mkdir -p venv-dir && python3 -m venv venv-dir
```
Comment (kbieganski) [inline action.yml:71]:
```suggestion
      run: cd $GITHUB_ACTION_PATH && mkdir -p venv-dir && python3 -m venv venv-dir
```

File: action.yml [hunk 3]
```diff
@@ -68,12 +68,12 @@ runs:
...
       shell: bash
 
     - id: test
       run: |
-        cd ${{github.action_path}} && sudo python3 action/run-in-renode.py \
+        cd ${GITHUB_ACTION_PATH} && sudo python3 action/run-in-renode.py \
         "${{ inputs.renode-run }}"                                         \
         "${{ inputs.devices }}"                                            \
         "${{ inputs.python-packages }}"                                    \
```
Comment (kbieganski) [inline action.yml:79]:
```suggestion
        cd $GITHUB_ACTION_PATH && sudo python3 action/run-in-renode.py \
        "${{ inputs.renode-run }}"                                     \
        "${{ inputs.devices }}"                                        \
        "${{ inputs.python-packages }}"                                \
```

File: action.yml [hunk 4]
```diff
@@ -38,12 +38,12 @@ runs:
...
         fi
-        if [ -f "$IMAGE" ] && [ $(realpath "$IMAGE") == "$(realpath ${{github.action_path}}/images.tar.xz)" ]; then
+        if [ -f "$IMAGE" ] && [ $(realpath "$IMAGE") == "$(realpath ${{ github.action_path }}/images.tar.xz)" ]; then
           echo File already exists;
         elif [ -f "$IMAGE" ]; then
-          cp "$IMAGE" ${{github.action_path}}/images.tar.xz;
+          cp "$IMAGE" ${{ github.action_path }}/images.tar.xz;
         else
-          wget -q --no-verbose "$IMAGE" -O ${{github.action_path}}/images.tar.xz;
+          wget -q --no-verbose "$IMAGE" -O ${{ github.action_path }}/images.tar.xz;
```
Comment (kbieganski) [inline action.yml:46]: So does `${{ github.action_path }}` work here? Or should all its occurrences be changed to `$GITHUB_ACTION_PATH`? I've not used GitHub Actions containers, but I think it should be the latter? Though I guess I shouldn't expect any GHA feature to be intuitive :)

**Review by kbieganski — CHANGES_REQUESTED** [review]
Thank you for the contribution! I left a few comments.

At some point we should add a test for containerized workflows. Although, to be honest, it's a bit silly that action maintainers have to worry about action-specific variables not always being available.

**Comment by WiktorOgrodnik** [comment]
@flynneva  I looked at the tests on your repository. The tests on jammy are probably failing because there is no bootloader, kernel and rootfs in an expected directory ./images/*. I think we need to change the `github.action_path` to `GITHUB_ACTION_PATH` everywhere, especially in the `get-images` step.

On focal there is another problem because it has Python in version 3.8.2, but parts of our type hints require at least Python 3.9 due to change PEP 585 (https://peps.python.org/pep-0585/). So you need to upgrade your Docker container to run this action properly. Perhaps we should note this requirement somewhere.

**Comment by WiktorOgrodnik** [comment]
@flynneva I have fixed the rest of ${{ github.action_path }} appearances in the pull request #25 if you want to see. I think this should fix one of your issues.

**Comment by flynneva** [comment]
@WiktorOgrodnik awesome stuff! I'll check it out probably over the weekend when I get a chance and report back 🙏🏼👍

**Pull request closed without merging**